proposals/001_interop.md

```diff
@@ -0,0 +1,473 @@
+claim to be authoritative publishers of any Docmaps, i.e., if `GET /docmap/{id}` returns any
+`publisher.id` which has as a prefix the `api_url` of this server. This endpoint MUST return
+valid results for each of those IDs.
+
+If present, MUST return a JSON body that is a decodable Docmap Publisher. MAY include a
+`@context` key for users to interpret as a JSON-LD object. MUST NOT return a top-level
+`@graph` key.
+
+```json
+ {
+   "@context": "https://w3id.org/docmaps/context.jsonld",
+   "name": "John Doe",
+   "id": "https://example.com/ex/docmaps/v1/publisher/example_press",
+   "homepage": "https://example.com/",
+   "logo": "https://example.com/logo.img",
+   "account": {
+     "id": "https://github.com/docmaps-project",
+     "service": "https://github.com"
+   }
+ },
+```
+
+where
+
+PATH | DETAILS
+| - | - |
+| `@context` |  If present, MUST be a JSON-LD context. SHOULD be exactly `"https://w3id.org/docmaps/context.jsonld"`  |
+| `id` |  The ID of the publisher. MUST be an IRI and MUST be a URL which when queried responds with this publisher exactly. |
+
+Publishers are always named nodes (nodes with IRIs for named nodes (nodes with IRIs) in the Docmaps specification. Therefore
+they are always considered persistent. Any signatory information is separate from  their RDF bodies (the data in this  endpoint's
+response), so those bodies can safely  change. However, if the IRI of a publisher changes, clients will need to handle the migration
+by their own logic until  the core Docmaps protocol is improved ([see Github  Issue here](https://github.com/Docmaps-Project/docmaps/issues/69)).
+That  improvement is out of the scope of this RFC.
+
+The authors of this specification anticipate the relationship between a server and specific Publishers whose signing identities
+are owned by the server may be expressed in extensions to this endpoint. See "Security" below.
+
+### Searching for docmaps based on content matter
+
+These endpoints are REQUIRED for all standard Docmaps servers. They enable consumers to discover what docmaps or objects
+can be retrieved from this server. They return information about further quueries that can be performed to retrieve docmaps.
+
+#### POST /search
+
+The search query MUST be accepted in the request body as type `application/json`. Although servers are invited to
+decide on search semantics that are most appropriate to their use case, servers MUST support the following body
+fields in the search:
+
+```json
+{
+  "query_terms": [
+    {
+      "match": "https://data-hub-api.elifesciences.org/enhanced-preprints/docmaps/v1/index",
+      "paths": ["publisher.id"]
+    },
+    {
+      "match": "docmap",
+      "paths": ["type"]
+    }
+  ]
+}
+```
+
+where
+
+PATH | DETAILS
+| - | - |
+| `query_terms` | MUST be an array of object. |
+| `query_terms[].match` | MUST be a full IRI or a JSON-LD shortcut present in the Docmaps JSON-LD context. |
```

Comment:
Shouldn't that allow any value valid for the given JSON path?
e.g. we are looking up values by an internal manuscript id (e.g. "12345") present in the DocMap in a nested field, not an IRI.

Reply:
A DOI seems to be another example.